snapcraft/plugins/script.py

```diff
@@ -0,0 +1,106 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The script plugin is useful for building script based parts.
+Script based projects use bash commands which drive the build.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keyword:
+    - script-name:
+      (string)
+      Run the given script name.
+    - stages:
+      (list of strings)
+      Run the script in the given stages. (default: ['build'])
```

Comment:
I understand what you're saying, but I still think it wouldn't hurt to at least have a simple bash plugin to base custom plugins from. Also I would imagine that running one command for the building phase and maybe another command for the pulling phase covers the majority of the use cases for bash based project building.